### PR TITLE
fix: removing generic typing from typeddata associated with eip712

### PR DESCRIPTION
**Pull request by jeanlarrieux**
`HSM` is now expecting a stringify JSON instead of JSON object. Also need `messageEncoded` to be a hex string of the Buffer instead of just a string represenation, as string representation leads to issues recreating the Buffer in `WP` due to encoding.

Ticket: BG-63009

### CODE REVIEW

File: modules/sdk-coin-eth/src/eth.ts [hunk 1]
```diff
@@ -2096,15 +2096,16 @@ export class Eth extends BaseCoin {
...
     if (version === SignTypedDataVersion.V1) {
       throw new Error('SignTypedData v1 is not supported due to security concerns');
     }
-
-    const sanitizedData = TypedDataUtils.sanitizeData(typedData.typedDataRaw);
+    const typedDataRaw = JSON.parse(typedData.typedDataRaw);
+    const sanitizedData = TypedDataUtils.sanitizeData(typedDataRaw as unknown as TypedMessage<any>);
     const parts = [Buffer.from('1901', 'hex')];
     const eip712Domain = 'EIP712Domain';
+    console.log('jeannius typedDataRaw: ', typedDataRaw);
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-eth/src/eth.ts:2108]: console.log

File: modules/sdk-coin-eth/src/eth.ts [hunk 2]
```diff
@@ -2096,15 +2096,16 @@ export class Eth extends BaseCoin {
    * @param typedData the typed data to prepare
    * @return a buffer of the result
    */
-  encodeTypedData(typedData: TypedData<never>): Buffer {
+  encodeTypedData(typedData: TypedData): Buffer {
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-eth/src/eth.ts:2099]: Changes to a required parameters on a public method is a breaking change. Needs to be documented in its' own commit for `sdk-coin-eth` separate from other modules.

**Review by mmcshinsky-bitgo — CHANGES_REQUESTED** [review]

File: modules/sdk-coin-eth/src/eth.ts [hunk 3]
```diff
@@ -2096,13 +2096,13 @@ export class Eth extends BaseCoin {
    * @param typedData the typed data to prepare
    * @return a buffer of the result
    */
-  encodeTypedData(typedData: TypedData<never>): Buffer {
```
Comment (olibeyene) [inline modules/sdk-coin-eth/src/eth.ts:2099]: this is a breaking change

File: modules/sdk-core/src/bitgo/utils/tss/baseTypes.ts [hunk 4]
```diff
@@ -119,8 +119,8 @@ export interface PopulatedIntentForMessageSigning extends PopulatedIntentBase {
   custodianMessageId?: string;
 }
 
-export interface PopulatedIntentForTypedDataSigning<T extends MessageTypes> extends PopulatedIntentBase {
-  messageRaw: TypedMessage<T>;
+export interface PopulatedIntentForTypedDataSigning extends PopulatedIntentBase {
+  messageRaw: string;
```
Comment (olibeyene) [inline modules/sdk-core/src/bitgo/utils/tss/baseTypes.ts:123]: this is also a breaking change

File: modules/sdk-core/src/bitgo/wallet/wallet.ts [hunk 5]
```diff
@@ -2788,7 +2788,7 @@ export class Wallet implements IWallet {
           intentType: 'signMessage',
           isTss: true,
           messageRaw: params.message.messageRaw,
-          messageEncoded: params.message?.messageEncoded ?? '',
+          messageEncoded: Buffer.from(params.message?.messageEncoded ?? '').toString('hex'),
```
Comment (olibeyene) [inline modules/sdk-core/src/bitgo/wallet/wallet.ts:2791]: if params.message?.messageEncoded is null, you should throw an error

**Review by olibeyene — CHANGES_REQUESTED** [review]
most of these change are breaking changes. please follow the developer relations guideline

File: modules/sdk-core/src/bitgo/wallet/wallet.ts (hunk 5 above)
Comment (jeanlarrieux) [reply modules/sdk-core/src/bitgo/wallet/wallet.ts:2791]: well the point was that future evm chains may not have encoding for message, hence why I made `messageEncoded` optional. If we are to throw an error, might as well make it mandatory.

Comment (olibeyene) [reply modules/sdk-core/src/bitgo/wallet/wallet.ts:2791]: gotcha

File: modules/sdk-coin-eth/src/eth.ts (hunk 3 above)
Comment (olibeyene) [reply modules/sdk-coin-eth/src/eth.ts:2099]: resolving this request since this feature are not live even though they are breaking changes

**Review by olibeyene — DISMISSED** [review]

**Review by mmcshinsky-bitgo — APPROVED** [review]

**Review by olibeyene — APPROVED** [review]

**Pull request merged**